wrappers.rs

```diff
@@ -0,0 +1,38 @@
+//! This module helps suppress two kinds of warnings: `deprecated` and `unstable_name_collisions`.
+//! New items are created that are noop-wrappers of the original items.
+//! The items' original paths are preserved.
```

Comment:
I'd prefer that we didn't create a top-level wrappers.rs file, and instead `#[allow(deprecated, unstable_name_collisions)]` as needed.

Reply:
Just a bit for you to consider, @jswrenn. Other than the sheer number of `#[allow()]`s, some of these warnings are in expressions that include multiple chained method calls. So the span of the `#[allow()]` is more than we intend. Feels clumsy to me.
Unless, of course, we extract the very call to which it is intended, which gets ugly immediately.

Call it and we'll do it.

Reply:
I think we should blanket-allow both `deprecated` and `unstable_name_collisions`.

For `deprecated`:
- We don't care when it's our own deprecated methods.
- We only take a dependency on one other crate, and I'm not particularly worried about it deprecating anything.
- We don't really care when the standard library deprecates methods. It's rare, and the standard library doesn't ever remove deprecated methods.

For `unstable_name_collisions`: This is a non-actionable warning at this time. Until [RFC 2845](https://github.com/rust-lang/rfcs/pull/2845) is implemented, both itertools and the standard library is deadlocked on moving forward. We can't remove methods that trigger `unstable_name_collisions` without breaking our dependents, and the standard library can't stabilize unstable conflicting methods without breaking itertools dependants.

Reply:
Sounds like you've thought it through. Will do.

Reply:
Done.